### PR TITLE
gh-114099: Add documentation for iOS platform

### LINKED ISSUE #115977: Remove Emscripten from the docs
Since Emscripten is no longer a tier 3 platform we should probably remove mentions of it from the docs.

### PR DESCRIPTION
**Pull request by freakboy3742**
Part of the PEP 730 work to add iOS support.

Adds an iOS usage guide (including documentation of how to add iOS to your own Xcode project), and flags modules in the standard library that aren't available on iOS.

Fixes #115977.

----
📚 Documentation preview 📚: https://cpython-previews--117057.org.readthedocs.build/

* Issue: gh-114099

### CODE REVIEW

**Comment by freakboy3742** [comment]
The linting check is currently picking up a missing link to `platform.ios_ver()`; this method is added (and [documented](https://github.com/python/cpython/pull/117052/files#diff-ebddb0d5cedd8047769d7907a0130eefa4cb9ab7bd0acc8fc84f0b13213d5a47R253)) in #117052.

File: Doc/includes/wasm-ios-notavail.rst
Comment (hugovk) [inline Doc/includes/wasm-ios-notavail.rst]: Re: https://github.com/python/cpython/issues/115977, we can remove all the Emscritpen mentions.

At least don't add new ones. 

Optionally, remove the others in edited lines, but fine to leave those be if you'd prefer to keep this PR focused.

Comment (freakboy3742) [reply Doc/includes/wasm-ios-notavail.rst]: I've purged *almost* all the references to Emscripten; there are some remaining references in historical release notes, plus in `library/sys.rst` and `library/intro.rst`. The remaining references are for behavior that still *exists* in the codebase, even if Emscripten isn't covered by Tier 3. AFAICT this is analogous to methods for Android support that are *documented*, even though Android isn't Tier 3 (... yet 😄). Where the text referred to specific platform descriptors, I've simplified to just say "on WebAssembly" (or similar).

File: Doc/library/dbm.rst
Comment (serhiy-storchaka) [inline Doc/library/dbm.rst]: I do not think that this is the right place for insertion.

File: Doc/library/curses.rst
Comment (serhiy-storchaka) [inline Doc/library/curses.rst]: Should not it be moved after the following paragraph?

File: Doc/using/iOS.rst
Comment (serhiy-storchaka) [inline Doc/using/iOS.rst]: There is no `platform.ios_ver()`.

Comment (freakboy3742) [reply Doc/using/iOS.rst]: See my comment [here](https://github.com/python/cpython/pull/117057#issuecomment-2008671157).

**Review by serhiy-storchaka — APPROVED** [review]
I am not an iOS expert, and cannot verify that all written about the iOS build is correct, but all other changes LGTM.

File: Doc/using/iOS.rst
Comment (JacobCoffee) [inline Doc/using/iOS.rst]:
```suggestion
The Python standard library has some notable omissions and restrictions on
```

File: Doc/library/os.rst
Comment (hugovk) [inline Doc/library/os.rst:42]: Tip: no need to rewrap when editing text, in fact it's easier to review if it's not been rewrapped because the line diff is easier.

Not yet in the devguide: https://github.com/python/devguide/issues/1018

No need to redo these, just a tip for next time :)

File: Doc/using/configure.rst
Comment (hugovk) [inline Doc/using/configure.rst]: We have a Sphinx role for linking to source files:

```suggestion
See :source:`iOS/README.rst`.
```

Could fix the `Mac/README.rst` above too.

File: Doc/library/intro.rst
Comment (hugovk) [inline Doc/library/intro.rst]:
```suggestion
  experience, such as :program:`pip`. To add Python code to your iOS app, you must use
```

re: https://devguide.python.org/documentation/markup/#roles

File: Doc/using/iOS.rst
Comment (hugovk) [inline Doc/using/iOS.rst]:
```suggestion
of that computer. Users then interact with Python by running a :program:`python`
```

File: Doc/using/iOS.rst
Comment (hugovk) [inline Doc/using/iOS.rst]:
```suggestion
run a :program:`python` executable, or interact with a Python REPL.
```

File: Doc/using/iOS.rst
Comment (hugovk) [inline Doc/using/iOS.rst]:
```suggestion
the ``Frameworks`` folder.
```

File: Doc/using/iOS.rst
Comment (hugovk) [inline Doc/using/iOS.rst]:
```suggestion
poses two problems:
```

File: Doc/using/iOS.rst
Comment (hugovk) [inline Doc/using/iOS.rst]:
```suggestion
Adding iOS to an Xcode project
```

File: Doc/using/iOS.rst
Comment (hugovk) [inline Doc/using/iOS.rst]: ?
```suggestion
Platform when you first run Xcode. Alternatively, you can add an iOS Simulator
Platform by selecting from the Platforms tab of the Xcode Settings panel.
```

File: Doc/using/iOS.rst
Comment (hugovk) [inline Doc/using/iOS.rst]:
```suggestion
     * User Script Sandboxing: No
```

File: Doc/using/iOS.rst
Comment (hugovk) [inline Doc/using/iOS.rst]:
```suggestion
     * Quoted Include In Framework Header: No
```

File: Doc/using/iOS.rst
Comment (hugovk) [inline Doc/using/iOS.rst:1]: Let's use a lowercase filename, all the others are (except blurbs, README, CONTRIBUTING), and the mixed-case is output as lowercase `ios.html` anyway.

File: Doc/using/index.rst
Comment (hugovk) [inline Doc/using/index.rst]:
```suggestion
   ios.rst
```

File: Misc/NEWS.d/next/Documentation/2024-03-20-12-41-47.gh-issue-114099.ad_Ck9.rst
Comment (hugovk) [inline Misc/NEWS.d/next/Documentation/2024-03-20-12-41-47.gh-issue-114099.ad_Ck9.rst]: The changelog is mostly present tense:
```suggestion
Add an iOS platform guide, and flag modules not available on iOS.
```

File: Misc/NEWS.d/next/Documentation/2024-03-20-15-12-37.gh-issue-115977.IMLi6K.rst
Comment (hugovk) [inline Misc/NEWS.d/next/Documentation/2024-03-20-15-12-37.gh-issue-115977.IMLi6K.rst]:
```suggestion
Remove compatibilty references to Emscripten.
```

File: Doc/using/configure.rst
Comment (freakboy3742) [reply Doc/using/configure.rst]: Will do.

File: Doc/using/iOS.rst
Comment (freakboy3742) [reply Doc/using/iOS.rst:1]: /me twitches in Apple branding guide... :-)

Will do.

**Review by hugovk — APPROVED** [review]
Thank you!

---

> The linting check is currently picking up a missing link to `platform.ios_ver()`; this method is added (and [documented](https://github.com/python/cpython/pull/117052/files#diff-ebddb0d5cedd8047769d7907a0130eefa4cb9ab7bd0acc8fc84f0b13213d5a47R253)) in #117052.

We'll have to merge this after that one. 👍

File: Doc/using/ios.rst
Comment (mhsmith) [inline Doc/using/ios.rst]:
```suggestion
Adding Python to an iOS project
-------------------------------
```

File: Doc/using/ios.rst
Comment (mhsmith) [inline Doc/using/ios.rst]:
```suggestion
Python to your project <adding-ios>`.
```

**Review by ned-deily — APPROVED** [review]
LGTM. We can merge after #117052 is merged.

**Pull request merged**
